subcommands/common_config.go

```diff
@@ -78,12 +78,16 @@ type LogConfigsOptions struct {
 	ListContFunc  func(string) (*client.DeviceConfigList, error)
 }
 
-func (opts *LogConfigsOptions) GetUsersLookupById(api *client.Api, factory string) {
+func GetUsersLookupById(api *client.Api, factory string) (map[string]client.FactoryUser, error) {
```

Comment:
Just IMHO this needs to be a helper method of the client itself, it is generic enough and has nothing specific for the config.

So that one could call it like `api.GetUsersMapById(factory)`.
If you go this way, then I think your idea to return an error from it (rather than dying) makes sense.